lib/core/uri_parser.js

```diff
@@ -416,6 +415,24 @@ function parseQueryString(query, options) {
     console.warn('Unsupported option `wtimeout` specified');
   }
 
+  if (
```

Comment:
My interpretation of the spec tests that "are designed to exercise the connection string parsing related to read concern and write concern" was that we need to use `parseConnectionString` to verify that our uri parser correctly parses the different concerns. With that assumption, I needed to add this code to the parser so the errors would be thrown here. I can't just call `WriteConcern.fromOptions(result)` here because that would create a circular dependency. To resolve that, I've tried to refactor `WriteConcern.fromUrl` to not rely on `parseConnectionString`, but I haven't been successful.